kep/260-network-aware-scheduling/README.md

```diff
@@ -0,0 +1,1432 @@
+                            x-kubernetes-int-or-string: true
+                          maxNetworkCost:
+                            type: integer
+                            default: 0
+                            minimum: 0
+                            maximum: 10000
+                            format: int64
+                            description: The max Network Cost between two workloads.
+                        required:
+                        - workloadName
+                        type: object
+                      type: array
+                  required:
+                  - workloadName
+                  - workloadRef
+                  type: object
+                type: array
+            required:
+            - numMembers
+            - topologySortingAlgorithm
+            - workloads
+            type: object
+          status:
+            description: Record workload allocations (Workload Name, replica id, hostname).
+            properties:
+              runningWorkloads:
+                description: The number of actively running workloads (e.g., pods).
+                format: int32
+                type: integer
+                minimum: 0
+              scheduleStartTime:
+                description: ScheduleStartTime of the AppGroup
+                format: date-time
+                type: string
+              topologyCalculationTime:
+                description: topologyCalculationTime of the AppGroup
+                format: date-time
+                type: string
+              topologyOrder:
+                description: The optimal order to schedule workloads on this App Group based on a given algorithm.
+                items:
+                  description: Workload Name and Priority index
+                  properties:
+                    workloadName:
+                      type: string
+                      description: Name of the Workload.
+                    index:
+                      type: integer
+                      default: 1
+                      minimum: 1
+                      format: int64
+                      description: Priority index for each workload (e.g., 1, 2, ...)
+                          (1 means workload should be scheduled first in the AppGroup)
+                  type: object
+                type: array
+            type: object
+        type: object
+    served: true
+    storage: true
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: []
+  storedVersions: []
+```
+
+### Example
+
+<p align="center"><img src="figs/chain.png" title="Chain" width="600" class="center"/></p>
+
+```yaml
+# Example App Group CRD spec
+apiVersion: scheduling.sigs.k8s.io/v1alpha1
+kind: AppGroup
+metadata:
+  name: a1
+spec:
+  numMembers: 3
+  topologySortingAlgorithm: KahnSort
+  workloads: 
+    - workloadName: P1 # Corresponds to the Kubernetes service (Label selector)
```

Comment:
Using a plain name in dependencies doesn't sound accurate, can we use the full ref spec, and also remove the workloadName field, sth. like this?

```yaml
workloads:
- workload:
    kind: Deployment
    apiVersion: apps/v1
    namespace: default
    name: P1
  dependencies:
  - workload:
      kind: Deployment
      apiVersion: apps/v1
      namespace: default
      name: P2
    minBandwidth: "100Mi"
    maxNetworkCost: 30
- workload:
    kind: Deployment
    apiVersion: apps/v1
    namespace: default
    name: P2

Reply:
Thanks for the suggestion @Huang-Wei. We have updated the AppGroup CRD and the API definitions.